daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -410,6 +411,12 @@ object ImmArray {
     override def copyToArray[B >: A](xs: Array[B], dstStart: Int, dstLen: Int): Unit =
       array.copyToArray(xs, dstStart, dstLen)
 
+    override def map[B, That](f: A => B)(implicit bf: CanBuildFrom[ImmArraySeq[A], B, That]): That =
+      bf match {
+        case _: IASCanBuildFrom[B] => array.map(f).toSeq
+        case _ => super.map(f)(bf)
+      }
+
```

Comment:
@S11001001 what about `flatMap`.. we should also override it, applying the same approach.

Reply:
Not really; it can't quite benefit from this in the same way. We don't know the size of the output for flatMap, and the default implementation is already quite well-optimized for building a result ImmArraySeq of unknown size.

Now, if you're talking about things where we know the output size, like `++`? Absolutely. The trick is "is there a more optimal implementation _using the specific properties of ImmArray_", as the ones we already have are quite good for efficiently indexable underlying stores that can be built up element-by-element from left to right. For example, there wouldn't be any point to collectFirst, or collect, and even the `iterator` that's there, we probably shouldn't have bothered.